src/library/Cardano/Types.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-- |
+-- Copyright: © 2018-2020 IOHK
+-- License: Apache-2.0
+--
+module Cardano.Types
+    (
+    -- * Coin
+      Coin (..)
+    , coinIsValid
+
+    -- * UTxO
+    , UTxO (..)
+    , utxoBalance
+    , utxoPickRandom
+
+    ) where
+
+import Prelude
+
+import Control.DeepSeq
+    ( NFData (..) )
+import Crypto.Number.Generate
+    ( generateBetween )
+import Crypto.Random.Types
+    ( MonadRandom )
+import Data.Map.Strict
+    ( Map )
+import Data.Word
+    ( Word64 )
+import Fmt
+    ( Buildable (..), blockListF' )
+import GHC.Generics
+    ( Generic )
+import Numeric.Natural
+    ( Natural )
+import Quiet
+    ( Quiet (Quiet) )
+
+import qualified Data.Map.Strict as Map
+
+{-------------------------------------------------------------------------------
+                                     Coin
+-------------------------------------------------------------------------------}
+
+-- | A non-negative integer value that represents a number of Lovelace.
```

Comment:
I suspect this should be *positive*, rather than *non-negative*. I've created an issue here: https://github.com/input-output-hk/cardano-coin-selection/issues/38

Reply:
for me non-negative is precise. it can be `>= 0` like encoded in `Word64` . no ?

Reply:
I mentioned it here as the specification defines it as a positive integer:

https://github.com/input-output-hk/cardano-coin-selection/pull/16/files#diff-f8a2c3e44e2d8bf1d848c3c112859487R211

We already impose an upper bound, which corresponds to 45 million billion Lovelace (45 followed by 15 zeros), which is significantly less than `maxBound :: Word64`.